src/error.rs

```diff
@@ -212,6 +216,14 @@ pub enum ErrorKind {
         file_path: String,
     },
 
+    #[cfg(feature = "tokio-runtime")]
+    #[error(display = "{}", _0)]
+    ReqwestError(#[error(source)] reqwest::Error),
```

Comment:
since these two errors only ever occur in the context of aws auth, I wonder if it would be better to just return them as authentication errors containing their error descriptions, rather than the full error structs themselves. This is similar to how we repackage any connection errors that occur during the authentication process as auth errors. 

Also, do we need to perform any sort of redaction of the contents of these errors?